api/app/signals/apps/signals/migrations/0025_create_history_view_in_db.py

```diff
@@ -0,0 +1,79 @@
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('signals', '0024_auto_20181031_1655'),
+    ]
+
+    operations = [
+        migrations.RunSQL("""
+DROP VIEW IF EXISTS "signals_history_view";
+CREATE VIEW "signals_history_view" AS
+    SELECT
+        CAST("identifier" AS varchar(255)),
+        _signal_id,
+        CAST("when" AS timestamptz),
+        CAST("what" AS varchar(255)),
+        CAST("who" AS varchar(255)),
+        CAST("extra" AS varchar(255)),
+        CAST("description" AS varchar(3000))  /* Most extreme case; the Note model's text field. */
+    FROM (
+        SELECT
+            CONCAT('UPDATE_STATUS_', CAST("s"."id" as VARCHAR(255))) AS "identifier",
```

Comment:
Waarom een samengesteld identifier. Is het nodig dat deze uniek is? Waarom ? De what samen oorspronkelijk ID is ook een uniek iets

Reply:
Ik ga kijken of Django ook tevreden is met een `unique_together` op het model.

Reply:
Dan kan je het ook gewoon zo laten. Zelf vind ik het een tekortkoming van Django dat er geen samengestelde primary keys mogelijk zijn.

Reply:
Het is inderdaad (op dit moment) niet mogelijk om in Django meerdere velden te markeren als primary key. De **unique_together** kan worden toegevoegd als dit nodig is, hierdoor maak je de velden die je opgeeft in ieder geval uniek. Echter blijf je dan zitten met het primary key probleem. Als deze niet gedefinieerd word dan zal Django deze zelf genereren. Naar mijn idee kan dit zo blijven zoals het nu is aangezien de identifier ook nodig is voor de front-end (zoals Jasper aangaf).